rust/xetblob/src/xet_repo.rs

```diff
@@ -334,6 +335,24 @@ impl XetRepoWriteTransaction {
     /// Opens a file a for write, returning a XetWFileObject
     /// which provides file write capability
     pub async fn open_for_write(&mut self, filename: &str) -> anyhow::Result<Arc<XetWFileObject>> {
+        // If appropriate, download the shard hints first
+        if let PFTRouter::V2(ref tr_v2) = &self.translator.pft {
+            if let Some(body) = self
+                .bbq_client
+                .perform_stat_query(self.remote_base_url.clone(), &self.branch, filename)
+                .await?
+            {
+                let ptr_file =
+                    PointerFile::init_from_string(&String::from_utf8_lossy(&body), filename);
+
+                if ptr_file.is_valid() {
+                    // TODO: strategies to limit this, and limit the number of shards downloaded?
+                    let file_hash = MerkleHash::from_hex(ptr_file.hash())?;
+                    tr_v2.fetch_hinted_shards_for_file(&file_hash).await?;
```

Comment:
Lets not do this right here. I worry that this will introducing a ton of overhead for large recursive uploads. Can we add this as a seperate method instead and plumb it through pyxet? It is easier to introduce more elaborate heuristics on the Python side. I am thinking:
 - user hinting
 - for large cps, we will do this.
 - for recursive cps we can poll the folder and fetch shards of large files in the same folder.
 - For xetfs file writes we could buffer X MB of writes in Python, then look for hints.

Or stuff like that.